astropy/io/fits/hdu/compressed/compressed.py

```diff
@@ -362,6 +362,19 @@ def __init__(
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+        # Handle case where ZBLANK is set on compressed integer HDUs by copying
+        # it to BLANK.
+        if self._bitpix > 0 and "ZBLANK" in self._header:
+            zblank = self._header["ZBLANK"]
+            orig_blank = self._header.get("BLANK", zblank)
+            if orig_blank != zblank:
+                raise ValueError(
+                    "Both ZBLANK and BLANK found in header for compressed image "
+                    "with different values "
+                    f"(ZBLANK={zblank}, BLANK={orig_blank})"
+                )
+            self._header["BLANK"] = zblank
```

Comment:
Not sure if all of this is needed, the comment is misleading since in the end the value BLANK value is used (if it is the same as ZBLANK). Also if ZBLANK is set in this case (integer array) it should probably not be trusted anyway.

Reply:
Sure, would you prefer to completely ignore ZBLANK for integer files instead of trying to work with it?

Reply:
Yes I think it should be ignored.

Reply:
Changes made as requested